pkg/expr/classic/classic.go

```diff
@@ -184,18 +176,45 @@ func (ccc *ConditionsCmd) Execute(ctx context.Context, vars mathexp.Vars) (mathe
 	return newRes, nil
 }
 
-// UnmarshalConditionsCmd creates a new ConditionsCmd.
-func UnmarshalConditionsCmd(rawQuery map[string]interface{}, refID string) (*ConditionsCmd, error) {
+// ConditionJSON is the JSON model for a single condition in ClassicConditionsCmd.
+// It is based on services/alerting/conditions/query.go's newQueryCondition().
+type ConditionJSON struct {
+	Evaluator ConditionEvalJSON     `json:"evaluator"`
+	Operator  ConditionOperatorJSON `json:"operator"`
+	Query     ConditionQueryJSON    `json:"query"`
+	Reducer   ConditionReducerJSON  `json:"reducer"`
+	// Params []interface{} `json:"params"` (Unused)
+}
+
+type ConditionEvalJSON struct {
+	Params []float64 `json:"params"`
+	Type   string    `json:"type"` // e.g. "gt"
+}
+
+type ConditionOperatorJSON struct {
+	Type string `json:"type"`
+}
+
+type ConditionQueryJSON struct {
+	Params []string `json:"params"`
+}
+
+type ConditionReducerJSON struct {
+	Type string `json:"type"`
+}
+
+// UnmarshalConditionsCmd creates a new ClassicConditionsCmd.
+func UnmarshalConditionsCmd(rawQuery map[string]interface{}, refID string) (*ClassicConditionsCmd, error) {
 	jsonFromM, err := json.Marshal(rawQuery["conditions"])
 	if err != nil {
 		return nil, fmt.Errorf("failed to remarshal classic condition body: %w", err)
 	}
-	var ccj []ClassicConditionJSON
+	var ccj []ConditionJSON
```

Comment:
I think we should pick either "ClassicCondition" or "Condition" and use them consistently through the package. ConditionJSON doesn't seem to be used by anything except classic conditions.

Reply:
Should we then change:

```
type ClassicConditionsCmd struct {
	Conditions []condition
	refID      string
}
```

to:

```
type ClassicConditionsCmd struct {
	Conditions []classicCondition
	refID      string
}
```

Reply:
I kept it as `Condition` because the package is called `classic`.